pkg/clusteragent/admission/mutate/agent_sidecar/README.md

```diff
@@ -0,0 +1,87 @@
+- Creating datadog secrets in every namespace where you wish to inject the agent sidecar
+
+This allows the user to customize the matching criteria for the webhook. It allows specifying which pods will be injected with the agent sidecar.
+
+With this configuration, the default agent sidecar will be injected, in addition to any overrides set by the user in the specified profiles.
+
+### Custom Selectors/Profiles Without Provider
+
+This configuration includes the following:
+- Enabling the feature
+- Setting custom selectors
+- Setting custom profiles
+- Setting a provider
+- Creating datadog secrets in every namespace where you wish to inject the agent sidecar
+
+This allows the user to customize the matching criteria for the webhook. It allows specifying which pods will be injected with the agent sidecar.
+
+With this configuration, the default agent sidecar will be injected, in addition to any overrides set by the user.
+
+Having set a provider, the agent sidecar will also get automatically the necessary configurations for the targeted provider.
+
+## Expected Behaviour
+
+The table below shows the expected behaviour when the feature is enabled.
+
+Note that currently we only support creating 1 selector and 1 profile (config override).
+Creating multiple selectors and/or overrides will result in not registering any webhook.
+
+| Custom Selectors / Profiles Set | Provider Set       | Provider Supported | Expected Behaviour                                                                                                                                      |
+|---------------------------------|--------------------|--------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------|
+| :x:                             | :heavy_check_mark: | :heavy_check_mark: | Agent sidecar should be injected on pods having the provider label key set (`agent.datadoghq.com/sidecar: <provider>`)                                  |
+| :x:                             | :heavy_check_mark: | :x:                | No agent sidecar should be injected, and an error message will be logged in the cluster agent:  "agent sidecar provider is not supported: foo-provider" |
+| :x:                             | :x:                | :x:                | No agent sidecar should be injected, and an error message will be logged in the cluster agent:  "agent sidecar provider is not supported"               |
+| :heavy_check_mark:              | :heavy_check_mark: | :heavy_check_mark: | The agent sidecar container should be injected only on pods matching the selector, and the `DD_EKS_FARGATE` label should be set to `true`               |
+| :heavy_check_mark:              | :heavy_check_mark: | :x:                | Agent sidecar should be injected, and you must find an error message in the cluster agent logs "unsupported provider: foo"                              |
```

Comment:
shouldn't be injected

Reply:
Why do you think so?

If user set a custom selector, pods matching the selector should be injected.

Having an unsupported provider should lead to failure in setting provider-specific overrides. But why should it lead to failure of injection?

Reply:
OK. I think it'll make more sense once there are more providers :+1: